test/test_exasollog.lua

```diff
@@ -0,0 +1,94 @@
+local luaunit = require("luaunit")
+local mockagne = require("mockagne")
+
+local when, verify, any = mockagne.when, mockagne.verify, mockagne.any
+local date_pattern = "%Y-%m-%d"
+
+test_log = {}
+
+function test_log:setUp()
+    self.today = os.date(date_pattern)
+    self.socket_mock = mockagne.getMock()
+    self.tcp_mock = mockagne.getMock()
+    self.client_mock = self.tcp_mock -- on connect(), the socket library promotes a TCP socket to a client socket
+    when(self.socket_mock.tcp()).thenAnswer(self.tcp_mock)
+    when(self.tcp_mock:connect(any(), any())).thenAnswer(1)
+    when(self.socket_mock.gettime()).thenAnswer(1000000)
+    package.preload["socket"] = function () return self.socket_mock end
```

Comment:
That looks like an integration tests?

Reply:
Not really. That's why I am mocking like crazy. I don't want to test the world around it.

Reply:
Usually an integration test would be a good option here, but there are too many variables (like timestamps) that make deterministic tests a pain. So I chose to mock foreign objects in this case. Something I try to avoid otherwise.